ui/components/TodoList/index.js

```diff
@@ -0,0 +1,166 @@
+import { useEffect, useState } from "react";
+import gql from "graphql-tag";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import Link from "next/link";
+import { CheckIcon } from "components/Icons";
+import HappySpinner from "components/HappySpinner";
+import NavItem from "components/Header/NavItem";
+
+const GET_TODO_INFO = gql`
+  query TodoInfo {
+    orgMembers(limit: 2) {
+      id
+    }
+
+    events(limit: 1) {
+      id
+    }
+  }
+`;
+
+const SET_TODOS_FINISHED = gql`
+  mutation SetTodosFinished {
+    setTodosFinished {
+      id
+      finishedTodos
+    }
+  }
+`;
+
+const LoadingBar = ({ ratio }) => {
+  return (
+    <div className="flex items-center mx-3 space-x-4 mb-6">
+      <div className="text-xs text-gray-500">{~~(ratio * 100)}%</div>
+      <div className="flex-1 bg-gray-200 rounded-full h-2">
+        <div
+          className="bg-purple-700 rounded-full h-2"
+          style={{ width: `${ratio * 100}%` }}
+        ></div>
+      </div>
+    </div>
+  );
+};
+
+const TodoItem = React.forwardRef(({ onClick, href, todo, index }, ref) => {
+  return (
+    <a
+      className={`flex space-x-2 rounded ${
+        todo.link ? "hover:bg-gray-300 cursor-pointer" : ""
+      } ${todo.done ? "opacity-60" : ""}`}
+      onClick={onClick}
+      href={href}
+      ref={ref}
+    >
+      <div className="flex-none rounded-full p-1 mt-1.5 mx-2 h-7 w-7 bg-gray-100 flex items-center justify-center">
+        {todo.done ? <CheckIcon /> : index + 1}
+      </div>
+      <div>
+        <div className="text-lg">{todo.title}</div>
+        <div className="text-sm text-gray-700">{todo.desc}</div>
+      </div>
+    </a>
+  );
+});
+
+const TodoList = () => {
+  const { data, loading, error } = useQuery(GET_TODO_INFO);
+  const [setTodosFinished] = useMutation(SET_TODOS_FINISHED);
+  const [loc, setLoc] = useState("");
+  const [allDone, setAllDone] = useState(false);
+
+  const rawTodos = [
+    {
+      title: "Create community",
+      desc: `This is your own home on the Plato platform, now available under ${loc}`,
```

Comment:
I think `loc` could be more simple: `${currentOrg.subdomain}.${process.env.DEPLOY_URL}`, to avoid using `useEffect` and relying on the `window` object which is only available on the client side render

Reply:
Yeah thanks that's much more reasonable

Reply:
fixed